src/argilla/training/setfit.py

```diff
@@ -0,0 +1,161 @@
+class ArgillaSetFitTrainer(ArgillaTransformersTrainer):
+    _logger = logging.getLogger("ArgillaTransformersTrainer")
+
+    require_version("torch")
+    require_version("datasets")
+    require_version("transformers")
+    require_version("setfit")
+
+    def __init__(self, dataset, record_class, multi_label: bool = False, model: str = None, seed: int = None):
+        if model is None:
+            model = "all-MiniLM-L6-v2"
+        super().__init__(dataset=dataset, record_class=record_class, multi_label=multi_label, model=model, seed=seed)
+
+    def init_args(self):
+        from setfit import SetFitModel, SetFitTrainer
+
+        self.setfit_model_kwargs = get_default_args(SetFitModel.from_pretrained)
+        self.setfit_model_kwargs["pretrained_model_name_or_path"] = self.model
+        self.setfit_model_kwargs["multi_target_strategy"] = self.multi_target_strategy
+        self.setfit_model_kwargs["device"] = self.device
+
+        self.setfit_trainer_kwargs = get_default_args(SetFitTrainer.__init__)
+        self.setfit_trainer_kwargs["column_mapping"] = self._column_mapping
+        self.setfit_trainer_kwargs["train_dataset"] = self._train_dataset
+        self.setfit_trainer_kwargs["eval_dataset"] = self._eval_dataset
+        self.setfit_trainer_kwargs["seed"] = self._seed
+
+        self._model = None
+
+    def update_config(
+        self,
+        **setfit_kwargs,
+    ):
+        """
+        Updates the `setfit_model_kwargs` and `setfit_trainer_kwargs` dictionaries with the keyword
+        arguments passed to the `update_config` function.
+        """
+        from setfit import SetFitTrainer
+
+        self.setfit_model_kwargs.update(setfit_kwargs)
+
+        self.setfit_trainer_kwargs.update(setfit_kwargs)
+        self.setfit_trainer_kwargs = filter_allowed_args(SetFitTrainer.__init__, **self.setfit_trainer_kwargs)
+
+    def __repr__(self):
+        formatted_string = []
+        arg_dict = {
+            "'SetFitModel'": self.setfit_model_kwargs,
+            "'SetFitTrainer'": self.setfit_trainer_kwargs,
+        }
+        for arg_dict_key, arg_dict_single in arg_dict.items():
+            formatted_string.append(arg_dict_key)
+            for key, val in arg_dict_single.items():
+                formatted_string.append(f"{key}: {val}")
+        return "\n".join(formatted_string)
+
+    def train(self, path: str = None):
+        """
+        We create a SetFitModel object from a pretrained model, then create a SetFitTrainer object with
+        the model, and then train the model
+        """
+        from setfit import SetFitModel, SetFitTrainer
+
+        self._model = SetFitModel.from_pretrained(**self.setfit_model_kwargs)
+        self.setfit_trainer_kwargs["model"] = self._model
+        self.__trainer = SetFitTrainer(**self.setfit_trainer_kwargs)
+        self.__trainer.train()
+        if self._eval_dataset:
+            self._metrics = self.__trainer.evaluate()
+        else:
+            self._metrics = None
```

Comment:
These are never used, is that right? Same for transformers.
This is what I get when training, by the way:
<details><summary>Training logs</summary>

```
[04/05/23 12:12:35] WARNING  WARNING:ArgillaTrainer:ArgillaBaseTrainer info:                                                                                               base.py:140
                             _________________________________________________________________
                             These baseline params are fixed:
                                 dataset: research_titles_multi_label
                                 task: DatasetForTextClassification
                                 multi_label: True
                                 train_size: 0.8
                                 seed: None

                             <class 'argilla.training.setfit.ArgillaSetFitTrainer'> info:
                             _________________________________________________________________
                             The parameters are configurable via `trainer.update_config()`:
                                 'SetFitModel'
                             force_download: False
                             resume_download: False
                             proxies: None
                             token: None
                             cache_dir: None
                             local_files_only: False
                             has_token: None
                             arg_name: None
                             arg_value: None
                             pretrained_model_name_or_path: all-MiniLM-L6-v2
                             multi_target_strategy: one-vs-rest
                             device: cpu
                             'SetFitTrainer'
                             model: None
                             train_dataset: Dataset({
                                 features: ['id', 'text', 'label', 'binarized_label'],
                                 num_rows: 14
                             })
                             eval_dataset: Dataset({
                                 features: ['id', 'text', 'label', 'binarized_label'],
                                 num_rows: 4
                             })
                             model_init: None
                             metric: accuracy
                             loss_class: <class 'sentence_transformers.losses.CosineSimilarityLoss.CosineSimilarityLoss'>
                             num_iterations: 20
                             num_epochs: 1
                             learning_rate: 2e-05
                             batch_size: 16
                             seed: 42
                             column_mapping: {'text': 'text', 'binarized_label': 'label'}
                             use_amp: False
                             warmup_proportion: 0.1
                             distance_metric: <function BatchHardTripletLossDistanceFunction.cosine_distance at 0x0000016D3312C048>
                             margin: 0.25
                             samples_per_label: 2

                             Using the trainer:
                             _________________________________________________________________
                             `trainer.train(output_dir)` to train to start training. `output_dir` is the directory to save the model automatically.
                             `trainer.predict(text, as_argilla_records=True)` to make predictions.
                             `trainer.save(output_dir)` to save the model manually.
config.json not found in HuggingFace Hub
                    WARNING  WARNING:huggingface_hub.hub_mixin:config.json not found in HuggingFace Hub                                                               hub_mixin.py:184
model_head.pkl not found on HuggingFace Hub, initialising classification head with random weights. You should TRAIN this model on a downstream task to use it for predictions and inference.
Applying column mapping to training dataset
***** Running training *****
  Num examples = 300
  Num epochs = 1
  Total optimization steps = 19
  Total train batch size = 16
Iteration: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 19/19 [00:06<00:00,  2.78it/s] 
Epoch: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:06<00:00,  6.85s/it] 
Applying column mapping to evaluation dataset
***** Running evaluation *****
```
</details>
The evaluation output is never printed or displayed.

Reply:
I fixed this.